scripts/translate_utils.py

```diff
@@ -3,9 +3,95 @@
 import shutil
 import yaml
 
+class TranslatableAsset:
+    translatable_attributes = []
+
+    def __init__(self, asset: dict):
+        self.asset = asset
+        for key in ASSET_FOLDER_MAPPING:
+            if key in asset:
+                self.asset_type = ASSET_FOLDER_MAPPING[key]
+                break
+
+    def extract_text(self):
+        """
+        Extract text from an asset.
+        """
+
+    def extract_text(self):
+        """
+        Remove any variables from the content which should be omitted.
+        """
+        strings = []
+        for var_path in self.translatable_attributes:
+            strings.extend(self.translate_var(self.asset, var_path.split(".")))
+
+        return strings
+
+    def translate_var(self, content, var_path: list):
+        """
+        Helper method to remove content from the content dict.
+        """
+        if not content:
+            return []
+        if len(var_path) == 1:
+            # If the content is a list, we need to iterate over each item in the list
+            if isinstance(content, list):
+                 strings = []
+                 for item in content:
+                     strings.append(item.get(var_path[0], ""))
+                 return strings
+             # print("Translating asset: ", self.asset_type, "var_path: ", var_path)
+            string = [content.get(var_path[0], "")]
+            return string or []
+        else:
+            # If the content is a list, we need to iterate over each item in the list
```

Comment:
Same comments as above, I'm confused if this is a list or not, or when it might be one instead of the other.